remoteworkitem/remoteworkitem_test.go

```diff
@@ -72,9 +93,95 @@ func TestGitHubIssueMapping(t *testing.T) {
 	}
 
 	for _, localWorkItemKey := range workItemMap {
-		t.Log("Mapped ", localWorkItemKey)
-		if workItem.Fields[localWorkItemKey] == nil {
+		t.Log("Mapping ", localWorkItemKey)
+		_, ok := workItem.Fields[localWorkItemKey]
+		if ok == false {
+			t.Error(fmt.Sprintf("%s not mapped", localWorkItemKey))
+		} else {
+			t.Log("Mapped to ", workItem.Fields[localWorkItemKey])
+		}
+	}
+}
+
+func TestJiraIssueMapping(t *testing.T) {
+
+	content, err := test.LoadTestData("jira_issue_mapping.json", provideRemoteJiraData)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	workItemMap := WorkItemKeyMaps[ProviderJira]
+	remoteWorkItem := RemoteWorkItem{ID: "xyz", Content: []byte(content)}
+
+	remoteWorkItemImpl := RemoteWorkItemImplRegistry[ProviderJira]
+	ji, err := remoteWorkItemImpl(remoteWorkItem)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+	workItem, err := Map(ji, workItemMap)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, localWorkItemKey := range workItemMap {
+		t.Log("Mapping ", localWorkItemKey)
+		_, ok := workItem.Fields[localWorkItemKey]
+		if ok == false {
 			t.Error(fmt.Sprintf("%s not mapped", localWorkItemKey))
+		} else {
+			t.Log("Mapped to ", workItem.Fields[localWorkItemKey])
+		}
+	}
+}
+
+func TestFlattenGithubResponseMap(t *testing.T) {
+	testString, err := test.LoadTestData("github_issue_mapping.json", provideRemoteGithubData)
```

Comment:
You need to use resource.require here to specify when to run this test. See other tests for example.